### PR TITLE
task-100 -- BE: Redo Passport and ClockIn

### PR DESCRIPTION
**Pull request by ChanM0**
# Description
Redid passport based on what mikkal did. So now login takes in email and password now. 
As of now ClockIn takes Date and Time parameters. 

Please include a summary of the change and which issue is fixed. 
Testing now can include http tests. Login is now fully tested. 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,24 +1,25 @@
 os:
-- linux
+  - linux
 
 language: php
 php:
-- '7.1'
+  - "7.1"
```
Comment (digitalsoba) [inline .travis.yml]: Change back to single quotes pls

File: app/Http/Requests/Auth/LoginRequest.php [hunk 2]
```diff
@@ -24,18 +24,18 @@ public function authorize()
...
 
     public function messages()
     {
         return [
-            'username.required' => 'Username is required!',
-            'username.email' => 'Username must be an email!',
-            'username.exists' => 'Invalid username or password.',
-            'password.required' => 'Invalid username or password.'
+            'email.required' => 'Username is required!',
+            'email.email' => 'Username must be an email!',
```
Comment (chippingcolt) [inline app/Http/Requests/Auth/LoginRequest.php]: is there a better way to name this?

File: app/Models/Passport/OauthRefreshToken.php [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Models\Passport;
+
+use Illuminate\Database\Eloquent\Model;
+
+class OauthRefreshToken extends Model
+{
+    // empty
+    protected $toble = 'oauth_refresh_tokens';
```
Comment (chippingcolt) [inline app/Models/Passport/OauthRefreshToken.php]: "toble"

File: database/seeds/DatabaseSeeder.php [hunk 4]
```diff
@@ -11,6 +11,8 @@ class DatabaseSeeder extends Seeder
      */
     public function run()
     {
+
+        // $this->call(PassportSeeder::class);
```
Comment (chippingcolt) [inline database/seeds/DatabaseSeeder.php]: is this needed?

**Review by chippingcolt — CHANGES_REQUESTED** [review]

File: app/Http/Requests/Auth/LoginRequest.php [hunk 5]
```diff
@@ -24,18 +24,18 @@ public function authorize()
...
     }
 
     public function messages()
     {
         return [
-            'username.required' => 'Username is required!',
-            'username.email' => 'Username must be an email!',
-            'username.exists' => 'Invalid username or password.',
-            'password.required' => 'Invalid username or password.'
+            'email.required' => 'Username is required!',
```
Comment (404tonynotfound) [inline app/Http/Requests/Auth/LoginRequest.php]: Why not 'Email is required' and 'Must use a valid email address'

File: app/User.php [hunk 6]
```diff
@@ -29,5 +29,6 @@ class User extends Authenticatable
         'password', 'remember_token', 'id', 'updated_at', 'created_at', 'program_id'
     ];
 
+
```
Comment (404tonynotfound) [inline app/User.php:32]: WHY?!

File: app/Http/Controllers/Api/TimeLog/ClockInDomain/Services/ClockInService.php [hunk 7]
```diff
@@ -17,16 +17,16 @@ public function __construct(ClockInLogicContract $clockInLogicUtility)
         $this->clockInLogicUtility = $clockInLogicUtility;
     }
 
-    public function clockIn(string $timeStamp): array
+    public function clockIn(string $date, string $time): array
     {
         $user = Auth::user();
         $userId = $user->id;
 
         $this->clockInLogicUtility->verifyUserHasNotYetTimeLogged($userId);
```
Comment (404tonynotfound) [inline app/Http/Controllers/Api/TimeLog/ClockInDomain/Services/ClockInService.php]: I feel like we might need to refactor verifyUserHasNotYetTimeLogged. You're checking if there's any record in the table where clock out is null, but should we check it only for the date theyre trying to clock in for?

File: tests/Unit/DomainValueObjectsTests/TimePuncherTest.php [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+class TimePuncherTest extends TestCase
+{
+    /**
+     * A basic test example.
+     *
+     * @return void
+     */
+    public function testExample()
+    {
+        $this->assertTrue(true);
```
Comment (404tonynotfound) [inline tests/Unit/DomainValueObjectsTests/TimePuncherTest.php]: beautiful test

**Review by 404tonynotfound — APPROVED** [review]

File: app/User.php (hunk 6 above)
Comment (ChanM0) [reply app/User.php:32]: My linter adds the space.

File: app/Http/Controllers/Api/TimeLog/ClockInDomain/Services/ClockInService.php (hunk 7 above)
Comment (ChanM0) [reply app/Http/Controllers/Api/TimeLog/ClockInDomain/Services/ClockInService.php]: True, I'll adjust that. Should we have a null check as a daily task process. If they any user forgot to clock out, notify them?

Comment (ChanM0) [reply app/Http/Controllers/Api/TimeLog/ClockInDomain/Services/ClockInService.php]: Or should we do that at the beginning of opening the app.

**Review by chippingcolt — APPROVED** [review]

**Pull request merged**